go/internal/converter_helpers.go

```diff
@@ -0,0 +1,124 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package internal
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/valkey-io/valkey-glide/go/v2/internal/errors"
+)
+
+// get type of T
+func GetType[T any]() reflect.Type {
+	var zero [0]T
+	return reflect.TypeOf(zero).Elem()
+}
+
+// ================================
+
+// convert (typecast) untyped response into a typed value
+// for example, an arbitrary array `[]any` into `[]string`
+type responseConverter interface {
+	convert(data any) (any, error)
+}
+
+// convert maps, T - type of the value, key is string
+type mapConverter[T any] struct {
+	next     responseConverter
+	canBeNil bool
+}
+
+// convert arrays, T - type of the value
+type arrayConverter[T any] struct {
+	next     responseConverter
+	canBeNil bool
+}
+
+// ================================
+
+// Converts an untyped map into a map[string]T
+func (node mapConverter[T]) convert(data any) (any, error) {
+	if data == nil {
+		if node.canBeNil {
+			return nil, nil
+		} else {
+			return nil, &errors.RequestError{Msg: fmt.Sprintf("Unexpected type received: nil, expected: map[string]%v", GetType[T]())}
```

Comment:
See error comments above. 

```suggestion
			return nil, fmt.Errorf("Unexpected type received: nil, expected: map[string]%v", GetType[T]())
```